controllers/operator/apigateway_controller.go

```diff
@@ -151,14 +155,19 @@ func (i *APIGatewayReconciler) reconcileFinalizer(ctx context.Context, apiGatewa
 	}
 
 	if apiGatewayCR.IsInDeletion() && hasFinalizer(apiGatewayCR) {
-		apiRulesCount, err := apiRulesCount(ctx, i.Client)
+		apiRulesFound, err := apiRulesExist(ctx, i.Client)
 		if err != nil {
 			return controllers.ErrorStatus(err, "Error during listing existing APIRules")
 		}
 
-		if apiRulesCount > 0 {
-			return controllers.WarningStatus(fmt.Errorf("could not delete API-Gateway CR since there are %d APIRule(s) present that block its deletion", apiRulesCount),
-				"There are APIRule(s) that block the deletion of API-Gateway CR. Please take a look at kyma-system/api-gateway-controller-manager logs to see more information about the warning")
+		oryRulesFound, err := oryRulesExist(ctx, i.Client)
+		if err != nil {
+			return controllers.ErrorStatus(err, "Error during listing existing ORY Oathkeeper Rules")
+		}
+
+		if apiRulesFound || oryRulesFound {
```

Comment:
Maybe we could also fail just after `apiRulesExist` so that the logs will only show the `Rule` error if no APIRule is present? It might not be obvious for the end user that some rules are created from APIRule

Reply:
I see your point. Will make some changes ✌🏻